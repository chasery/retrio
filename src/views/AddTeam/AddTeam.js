import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import Header from '../../components/Header/Header';
import Form from '../../components/Form/Form';
import FormField from '../../components/FormField/FormField';
// import Error from '../../components/Error/Error';
import './AddTeam.css';

function AddTeam(props) {
  const history = useHistory();
  const [teamName, setTeamName] = useState('');
  // const [error, setError] = useState(null);

  const handleAddTeam = (e) => {
    e.preventDefault();
  };

  return (
    <>
      <Header />
      <main role='main'>
        <section className='AddTeam'>
          <div className='AddTeam__wrapper'>
            <Form id='AddTeam' onSubmit={(e) => handleAddTeam(e)}>
              <div className='Form__header'>
                <h2>Add Team</h2>
              </div>
              <div className='Form__body'>
                <p>Add a SCRUM team to associate with Retrio boards.</p>
                <FormField
                  id='teamName'
                  label='Team Name'
                  type='text'
                  isRequired={true}
                  onChange={(e) => setTeamName(e.target.value)}
                  value={teamName}
                />
                {/* {error ? <Error message={error} /> : null} */}
                <div className='Form__controls'>
                  <button
                    className='Form__button secondary'
                    type='button'
                    onClick={() => history.push(`/teams`)}
                  >
                    Cancel
                  </button>
                  <button className='Form__button primary' type='submit'>
                    Add Team
                  </button>
                </div>
              </div>
            </Form>
          </div>
        </section>
      </main>
    </>
  );
}

export default AddTeam;
