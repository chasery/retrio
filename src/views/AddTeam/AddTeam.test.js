import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddTeam from './AddTeam';

describe('AddTeam component', () => {
  it('renders without error', () => {
    const wrapper = shallow(<AddTeam />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
