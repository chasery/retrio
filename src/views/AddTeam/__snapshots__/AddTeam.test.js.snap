// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTeam component renders without error 1`] = `
<Fragment>
  <Header />
  <main
    role="main"
  >
    <section
      className="AddTeam"
    >
      <div
        className="AddTeam__wrapper"
      >
        <Form
          id="AddTeam"
          onSubmit={[Function]}
        >
          <div
            className="Form__header"
          >
            <h2>
              Add Team
            </h2>
          </div>
          <div
            className="Form__body"
          >
            <p>
              Add a SCRUM team to associate with Retrio boards.
            </p>
            <FormField
              id="teamName"
              isRequired={true}
              label="Team Name"
              onChange={[Function]}
              type="text"
              value=""
            />
            <div
              className="Form__controls"
            >
              <button
                className="Form__button secondary"
                onClick={[Function]}
                type="button"
              >
                Cancel
              </button>
              <button
                className="Form__button primary"
                type="submit"
              >
                Add Team
              </button>
            </div>
          </div>
        </Form>
      </div>
    </section>
  </main>
</Fragment>
`;
